### PR TITLE
Luiz/op 419 crud subnames

### PR DESCRIPTION
**Pull request by luizmuraro**
closes [OP-419](https://linear.app/bleu-builders/issue/OP-419/crud-subnames)

## Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Performance improvement
- [ ] Code refactoring

## Description 

- Create, Read, Update and Delete subnames API
- Crete API and Route layer
- Added some basics validations using `zod`

## Test Instructions (if applicable)
You can test the endpoints using [this postman collection](https://github.com/user-attachments/files/17637956/Subname.API.postman_collection.json).

Next PR will contain a basic UI then it's possible to test it there too.

### CODE REVIEW

File: frontend/src/services/subname.service.ts [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+        name: data.name,
+        price: new Prisma.Decimal(data.price),
+        duration: data.duration,
+        maxSubnames: data.maxSubnames,
+      },
+    });
+
+    return {
+      ...pack,
+      price: Number(pack.price),
```
Comment (devjoaov) [inline frontend/src/services/subname.service.ts:32]: 👀 is this necessary?

File: frontend/src/services/subname.service.ts [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+        },
+      },
+      include: {
+        texts: true,
+        addresses: true,
+        subscriptionPack: true,
+      },
+    });
+
+    return {
```
Comment (devjoaov) [inline frontend/src/services/subname.service.ts:79]: why not just return the validated data or the created object?

File: frontend/src/app/api/subnames/route.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import { SubnameService } from "@/services/subname.service";
+import { NextResponse } from "next/server";
+
+export async function GET() {
+  try {
+    const subnameService = new SubnameService();
+
+    const subnames = await subnameService.getAllSubnames();
```
Comment (devjoaov) [inline frontend/src/app/api/subnames/route.ts:8]: we should paginate this

**Review by devjoaov — APPROVED** [review]
It looks great but I missed pagination for the subnames, imagine a pack with 100 subnames, it would be interesting to have paginated subnames

File: frontend/src/services/subname.service.ts [hunk 4]
```diff
@@ -0,0 +1,209 @@
+import { prisma } from "@/lib/prisma";
+import {
+  createSubnameSchema,
+  updateSubnameSchema,
+} from "@/schemas/subname.schema";
+import type {
+  CreateSubnameDTO,
+  SubnameResponseDTO,
+  SubscriptionPackResponseDTO,
+  UpdateSubnameDTO,
```
Comment (ribeirojose) [inline frontend/src/services/subname.service.ts:10]: couldn't we leverage zod/prisma types for this? seems like a bit of code duplication to have DTOs

File: frontend/src/app/api/subnames/route.ts (hunk 3 above)
Comment (ribeirojose) [reply frontend/src/app/api/subnames/route.ts:8]: perhaps this extension could be a good help? https://github.com/deptyped/prisma-extension-pagination

**Comment by luizmuraro** [comment]
Thanks for the reviews folks, changes are being made on https://github.com/bleu/namespace-offchain-manager/pull/7

**Pull request merged**